src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptDocumentNavigationServiceWrapper.cs

```diff
@@ -35,13 +35,25 @@ public bool TryNavigateToPosition(Workspace workspace, DocumentId documentId, in
         public bool TryNavigateToPosition(Workspace workspace, DocumentId documentId, int position, int virtualSpace, OptionSet? options, CancellationToken cancellationToken)
         {
             var obj = _underlyingObject;
-            return _threadingProvider.Service.Run(() => obj.TryNavigateToPositionAsync(workspace, documentId, position, virtualSpace, NavigationOptions.Default, cancellationToken));
+            return _threadingProvider.Service.Run(async () =>
+            {
+                var location = await obj.GetLocationForPositionAsync(
+                    workspace, documentId, position, virtualSpace, cancellationToken).ConfigureAwait(false);
+                return location != null &&
+                    await location.NavigateToAsync(NavigationOptions.Default, cancellationToken).ConfigureAwait(false);
```

Comment:
this is a genuine weird case.  the TS code lives at the feature layer, but has hte concept of navigation.  if we changes NavigateTo to be sync and require the UI thread, we'd have to introduce some sort of hack (likely through the IWorkspaceThreadingService) to somehow allow it ot move over.  I don't have a good idea of what we can do here that isn't hack.  Thoughts @jasonmalinowski ?

Reply:
Is there a reason the TypeScript stuff isn't at the editor features layer?